### PR TITLE
Fix race condition in bulk load and fast secondary key creation

### PR DESCRIPTION
**Pull request by lth**
Summary:
There is one special condition where alter table will not call
finalize_bulk_load on the handler that bulk loaded the primary key. Detect
this condition and call finalize_bulk_load before starting secondary key
creation.

Squash with: https://reviews.facebook.net/D61779

Test Plan: mtr

### CODE REVIEW

File: storage/rocksdb/ha_rocksdb.cc [hunk 1]
```diff
@@ -9811,7 +9811,45 @@ int ha_rocksdb::inplace_populate_sk(const TABLE* new_table_arg,
...
+    the destructor on the handler it is holding. Thus, its m_sst_info will not
+    be nullptr.
+
+    At this point, it is safe to refresh the snapshot because we know all other
+    open handlers have been closed at this point, and the one we're on is the
+    only one left.
+  */
+  if (m_sst_info != nullptr)
+  {
+    if ((res= finalize_bulk_load()))
```
Comment (vasilep) [inline storage/rocksdb/ha_rocksdb.cc:9846]: NIT: It's usually cleaner when assignment and test are done on separates lines.
NIT: One extra unneeded pair of ().

File: storage/rocksdb/ha_rocksdb.cc [hunk 2]
```diff
@@ -9811,7 +9811,45 @@ int ha_rocksdb::inplace_populate_sk(const TABLE* new_table_arg,
   const bool hidden_pk_exists = has_hidden_pk(table);
 
   int res= 0;
-  Rdb_transaction *tx;
+  Rdb_transaction *tx= get_or_create_tx(table->in_use);
+
+  /*
```
Comment (vasilep) [inline storage/rocksdb/ha_rocksdb.cc:9816]: I <3 the level of detail explaining WHY we are doing it.

File: storage/rocksdb/ha_rocksdb.cc (hunk 1 above)
Comment (alxyang) [reply storage/rocksdb/ha_rocksdb.cc:9846]: @vasilep we already have this assignment/test pattern all over the codebase in ha_rocksdb.cc, so might be best to remain consistent.  Also, the extra pair of () is necessary I believe, it's really doing
`if ((res= finalize_bulk_load()) != 0)`

**Comment by alxyang** [comment]
I'm good to accept.  The sooner this is in the better, so I can cherry-pick it into the next release (it's the last blocker).  @yoshinorim I remember you were saying you ran some long-executing tests this weekend, have they all passed?

It's passed all the cp_sql and data_correctness tests I've executed.

**Review by alxyang — APPROVED** [review]
@yoshinorim confirmed his long-running tests passed.

**Pull request closed without merging**